periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
- keep https://github.com/kubeflow/model-registry/pull/521
- keep https://github.com/kubeflow/model-registry/pull/523
- keep https://github.com/kubeflow/model-registry/pull/490
- keep https://github.com/kubeflow/model-registry/pull/533
- keep https://github.com/kubeflow/model-registry/pull/526
- keep https://github.com/kubeflow/model-registry/pull/530
- keep https://github.com/kubeflow/model-registry/pull/531
- keep https://github.com/kubeflow/model-registry/pull/532
- keep https://github.com/kubeflow/model-registry/pull/534
- keep https://github.com/kubeflow/model-registry/pull/538
- keep https://github.com/kubeflow/model-registry/pull/539
- keep https://github.com/kubeflow/model-registry/pull/540
- keep https://github.com/kubeflow/model-registry/pull/524

maintaining labels

lampajr commented:
@dependabot rebase

lampajr reviewed (approved):
lgtm

lampajr commented:
Hi @isinyaaa , I was expecting that python check would have been triggered in this PR as we are actually modifying the python workflow.

Created https://github.com/opendatahub-io/model-registry/pull/146 to fix this behavior, wdyt?

isinyaaa commented:
@dependabot rebase

dependabot[bot] committed:
build(deps-dev): bump @testing-library/jest-dom from 6.5.0 to 6.6.2 in /clients/ui/frontend (#521)

Bumps [@testing-library/jest-dom](https://github.com/testing-library/jest-dom) from 6.5.0 to 6.6.2.
- [Release notes](https://github.com/testing-library/jest-dom/releases)
- [Changelog](https://github.com/testing-library/jest-dom/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/jest-dom/compare/v6.5.0...v6.6.2)

---
updated-dependencies:
- dependency-name: "@testing-library/jest-dom"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jenny-s51 committed:
Applies bug fixes to UI screens (#523)

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

update search filter styling for archived models list

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fix tests

Signed-off-by: Jenny <32821331+jenny-s51@users.noreply.github.com>

fix tests

ederign committed:
feat(bff): Use envtest for kubernetes testing instead of hardcoded mock (#490)

* feat(bff): Use envtest for kubernetes testing instead of hardcoded mock

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

* feat(bff): using ginkgo and gomega as our test suite

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

* PR review from Griffin

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

---------

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

dependabot[bot] committed:
build(deps): bump github.com/onsi/gomega from 1.33.1 to 1.35.1 in /clients/ui/bff (#533)

Bumps [github.com/onsi/gomega](https://github.com/onsi/gomega) from 1.33.1 to 1.35.1.
- [Release notes](https://github.com/onsi/gomega/releases)
- [Changelog](https://github.com/onsi/gomega/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/gomega/compare/v1.33.1...v1.35.1)

---
updated-dependencies:
- dependency-name: github.com/onsi/gomega
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ederign committed:
Multiple versions bumps: patternfly (6.0.0), eslint-plugin-jsx-a11y (6.10.2), eslint-plugin-react (7.37.2) and http-proxy-middleware (cve) (#526)

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump @cypress/code-coverage from 3.13.4 to 3.13.5 in /clients/ui/frontend (#530)

Bumps [@cypress/code-coverage](https://github.com/cypress-io/code-coverage) from 3.13.4 to 3.13.5.
- [Release notes](https://github.com/cypress-io/code-coverage/releases)
- [Changelog](https://github.com/cypress-io/code-coverage/blob/master/.releaserc)
- [Commits](https://github.com/cypress-io/code-coverage/compare/v3.13.4...v3.13.5)

---
updated-dependencies:
- dependency-name: "@cypress/code-coverage"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump dompurify from 3.1.6 to 3.1.7 in /clients/ui/frontend (#531)

Bumps [dompurify](https://github.com/cure53/DOMPurify) from 3.1.6 to 3.1.7.
- [Release notes](https://github.com/cure53/DOMPurify/releases)
- [Commits](https://github.com/cure53/DOMPurify/compare/3.1.6...3.1.7)

---
updated-dependencies:
- dependency-name: dompurify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/brianvoe/gofakeit/v7 from 7.0.4 to 7.1.2 in /clients/ui/bff (#532)

Bumps [github.com/brianvoe/gofakeit/v7](https://github.com/brianvoe/gofakeit) from 7.0.4 to 7.1.2.
- [Release notes](https://github.com/brianvoe/gofakeit/releases)
- [Commits](https://github.com/brianvoe/gofakeit/compare/v7.0.4...v7.1.2)

---
updated-dependencies:
- dependency-name: github.com/brianvoe/gofakeit/v7
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/onsi/ginkgo/v2 from 2.19.0 to 2.21.0 in /clients/ui/bff (#534)

Bumps [github.com/onsi/ginkgo/v2](https://github.com/onsi/ginkgo) from 2.19.0 to 2.21.0.
- [Release notes](https://github.com/onsi/ginkgo/releases)
- [Changelog](https://github.com/onsi/ginkgo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/ginkgo/compare/v2.19.0...v2.21.0)

---
updated-dependencies:
- dependency-name: github.com/onsi/ginkgo/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Al-Pragliola committed:
core: openapi factories, do not require artifactType (#538)

* core: make server generation use default factories

..introducing those missing.

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* server generation

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* with non regression testing

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* linting

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* do not require artifactType

Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

* Update clients/python/tests/test_client.py

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

---------

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>
Signed-off-by: Isabella do Amaral <idoamara@redhat.com>
Co-authored-by: Matteo Mortari <matteo.mortari@gmail.com>
Co-authored-by: Isabella do Amaral <idoamara@redhat.com>

ederign committed:
build(deps): bump @typescript-eslint/parser from 8.9.0 to 8.12.2 in clients/ui/frontend (#539)

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

ederign committed:
Fixing slog.Logger.Error arg err should be a string or a slog.Attr (possible missing key or value) (#540)

Signed-off-by: Eder Ignatowicz <ignatowicz@gmail.com>

Griffin-Sullivan committed:
Add NotificationContext and useNotification hook (#524)

Signed-off-by: Griffin-Sullivan <gsulliva@redhat.com>

tarilabs committed:
Merge sync remote-tracking branch 'upstream/main' into tarilabs-20241108-sync

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>
